blog/content/post/2022-05-04-fobnail_provisioning.md

```diff
@@ -0,0 +1,292 @@
+  `keyEncipherment`.
+
+In any case, private key must never leave Fobnail Token. This may limit speed
+and maximal size of data that is signed or encrypted, as those tasks must be
+performed by the Token. For example, encrypting whole hard disk with Fobnail's
+public key directly is a bad idea, but you can use it to encrypt a key used for
+decrypting the drive.
+
+Obviously, support on Fobnail side is required to expose API that uses private
+key accordingly to the usage specified in the certificate, based on attestation
+result. This is not done in this phase.
+
+## Building and running
+
+Since we've changed the way EK certificate chain is obtained, Attester can no
+longer be run on simulated TPM as easy as before. Chain is now downloaded during
+platform provisioning, but for cases when Internet connection can't be trusted
+we plan to add option to [use certificate chain from file](https://github.com/fobnail/fobnail-attester/issues/37).
+
+Because TPM simulator is not used, Fobnail Token is also started separately.
+There is also a new application for Platform Owner, which should be started
+on a separate, secure machine, but for testing can also be run on the same
+system as the other components.
+
+### Platform Owner
+
+To building Platform Owner application, just run `make`:
+
+```shell
+$ git clone https://github.com/fobnail/fobnail-platform-owner.git --recurse-submodules
+$ cd fobnail-platform-owner
+$ make
+```
+
+It takes certificate chain and private key used for signing Fobnail Token's
+certificate on input:
+
+```shell
+$ ./bin/fobnail-platform-owner path/to/chain.pem path/to/key.priv
+```
+
+Proper key and chain can be generated by following [these instructions](https://fobnail.3mdeb.com/keys_and_certificates/#creating-platform-owner-certificate-chain-with-openssl).
+`chain.pem` is a concatenation of all certificates, starting with root:
+
+```shell
+$ cat root.crt ca1.crt ca2.crt > chain.pem
+```
+
+### Attester
+
+Attester is built similarly to Platform Owner:
+
+```shell
+$ git clone https://github.com/fobnail/fobnail-attester.git --recurse-submodules
+$ cd fobnail-attester
+$ make
+```
+
+It must be started as privileged user, in order to access TPM and SMBIOS:
+
+```shell
+$ sudo ./bin/fobnail-attester
+```
+
+Currently Attester must have physical NIC because its MAC is used as part of
+metadata. This is a known issue that makes it unusable on platforms that don't
+have one, such as virtual machines, wireless-only devices or controllers for
+which kernel doesn't have a driver.
+
+### Fobnail Token
+
+[Fobnail SDK](../2022-03-21-fobnail_3rd_phase#building-and-installing-sdk) must
+be installed before building Fobnail Token software. After that, clone Fobnail's
+repository:
+
+```shell
+$ git clone https://github.com/fobnail/fobnail.git --recurse-submodules
+$ cd fobnail
+```
+
+To build, flash and run on nRF52840:
+
+```shell
+$ ./build.sh FOBNAIL_PO_ROOT=path/to/PO/root.crt -t nrf --run
```

Comment:
Maybe we should mention here that FOBNAIL_PO_ROOT must point to path which is a subdirectory of Fobnail source directory (limitation of Docker).

Also, current command won't work, `FOBNAIL_PO_ROOT` must be set as env variable, not passed as arg.
```suggestion
$ env FOBNAIL_PO_ROOT=path/to/PO/root.crt ./build.sh -t nrf --run
```